integration/README.md

```diff
@@ -0,0 +1,37 @@
+# Integration tests
+
+This directory contains integration tests for the project.
+
+The tree structure generally mirrors the source code tree structure.
+
+Requirements for new files in this directory:
+* Every integration test file **must** use the `integration` build tag
+* Every package **must** be named after its directory with `_integration` appended
```

Comment:
> * Every integration test file **must** use the `integration` build tag
> * Every package **must** be named after its directory with `_integration` appended

Did you consider an alternative - do not add tag, modify 'make test' to ignore _integration packages? This would simplify requirements and minimize changes to the tests. What are downsides?

Reply:
I think we already able to run integration tests with 
`go test ./integration/...`

and all regular tests with
`go test ./libs/... ./cmd/... ./bundle/...`

no build tags necessary.

Reply:
The downside is that other test runners (e.g., manual calls of `go test`, `gotestsum`, or the VS Code test runner) will need to perform the same ignore to focus on unit tests.

I prefer having to be explicit because these tests will pick up whatever auth is configured (including your personal session) and start making API calls.

Reply:
Example from VS Code:
![Screenshot 2024-12-13 at 10 16 16](https://github.com/user-attachments/assets/81eac483-a6ac-4317-a843-047218c32f0f)

Reply:
We need some kind of marker to indicate they are integration tests. That was `CLOUD_ENV` previously, it is a build tag in this PR, and it can be something else as well, as long as it is consistently applied.

Changing the env vars we use is not really an option because we rely on the SDK to auth based on (the real) env vars. If we change this we'd need to pass them through in a different way.

Reply:
This is a possible approach we discussed - 
https://github.com/databricks/cli/commit/c26b724946b0fc868b056a255f22728b22a052f7

Reply:
Discussed in person.

Outcome: instead of build tags, we'll go with the env var approach to use as an integration test marker. But instead of adding a statement to every single test, we include a `main_test.go` in each package that performs this check.